src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -289,6 +289,11 @@ private ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e)
                 {
                     CheckFeatureAvailability(patternExpression, MessageID.IDS_FeatureNullPointerConstantPattern, diagnostics, patternExpression.Location);
                 }
+                else if (inputType.IsStringType())
```

Comment:
>inputType.IsStringType() [](start = 25, length = 24)

I don't believe this is correct. Please add a test where the `o` is of type `object` in `switch (o) { case $"test": break; }`

Reply:
It should be checking to see whether the pattern itself has a constant value of type string.

---
In reply to: [453030193](https://github.com/dotnet/roslyn/pull/45685#discussion_r453030193) [](ancestors = 453030193)